SwiftCompilerSources/Sources/Optimizer/DataStructures/InstructionRange.swift

```diff
@@ -181,3 +181,54 @@ struct InstructionRange : CustomStringConvertible, NoReflectionChildren {
     blockRange.deinitialize()
   }
 }
+
+extension InstructionRange {
+  enum PathOverlap {
+    case containsPath
```

Comment:
It would be nice to add some comments, e.g.
```
  // range:  ---
  //          |    pathBegin
  //          |       | 
  //          |     pathEnd
  //         ---
  case containsPath

  // range:  ---
  //          |    pathBegin
  //         ---       | 
  //                pathEnd
  // 
  case containsBegin
```